### PR TITLE
PHOENIX-6564 :- Add Phoenix Connection Failed Global Metric

### PR DESCRIPTION
**Pull request by lokiore**

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixMetricsIT.java [hunk 1]
```diff
@@ -1099,6 +1100,53 @@ public void testGetConnectionsThrottledForSameUrl() throws Exception {
...
+        } finally {
+            for (Connection c : connections) {
+                c.close();
+            }
+        }
+        assertEquals(1, GLOBAL_QUERY_SERVICES_COUNTER.getMetric().getValue());
+        assertEquals(1, GLOBAL_PHOENIX_CONNECTIONS_THROTTLED_COUNTER.getMetric().getValue());
+        assertEquals(3, GLOBAL_FAILED_PHOENIX_CONNECTIONS.getMetric().getValue());
+        assertTrue("Not all connections were attempted!",
+                attemptedPhoenixConnections <= GLOBAL_PHOENIX_CONNECTIONS_ATTEMPTED_COUNTER.getMetric().getValue());
```
Comment (dbwong) [inline phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixMetricsIT.java:1146]: Shouldn't we have equality here?

File: phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java [hunk 2]
```diff
@@ -254,163 +255,170 @@ private PhoenixConnection(ConnectionQueryServices services, String url,
...
-        this.isRunningUpgrade = isRunningUpgrade;
-        
-        this.logSamplingRate = Double.parseDouble(this.services.getProps().get(QueryServices.LOG_SAMPLE_RATE,
-                QueryServicesOptions.DEFAULT_LOG_SAMPLE_RATE));
-        if(isInternalConnection) {
-            GLOBAL_OPEN_INTERNAL_PHOENIX_CONNECTIONS.increment();
-        } else {
-            GLOBAL_OPEN_PHOENIX_CONNECTIONS.increment();
+            throw sqlException;
         }
```
Comment (dbwong) [inline phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java:421]: What about non-SQLExceptions that we might accidentally throw?

**Review by dbwong — COMMENTED** [review]
A couple of questions.

File: phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java (hunk 2 above)
Comment (lokiore) [reply phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java:421]: Yes I just added known SQL ones, forgot to add those, adding them.

File: phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixMetricsIT.java (hunk 1 above)
Comment (dbwong) [reply phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixMetricsIT.java:1154]: nit: Can we add a comment to the same?

**Comment by lokiore** [comment]
Thanks @dbwong can you do that, I don't have write access!

**Pull request merged**